### PR TITLE
[ll] vulkan: Add native X11 and Wayland integration support

### PR DESCRIPTION
**Pull request by davll**
Similar to PR #1518
Closes #1524

### CODE REVIEW

File: src/backend/vulkan/src/window.rs [hunk 1]
```diff
@@ -34,69 +38,155 @@ impl Drop for RawSurface {
...
+                dpy: display as *mut _,
+            };
+
+            unsafe { xlib_loader.create_xlib_surface_khr(&info, None) }
+                .expect("XlibSurface failed")
+        };
+
+        let (width, height) = unsafe {
+            use x11::xlib::{XGetWindowAttributes, XWindowAttributes};
+            use std::os::raw::{c_int, c_uint, c_ulong};
```
Comment (kvark) [inline src/backend/vulkan/src/window.rs]: these are not used

**Comment by kvark** [comment]
bors delegate=davll

**Comment by kvark** [comment]
@davll ping. Do you see this ready to merge?

**Comment by davll** [comment]
Yes, I rebased the PR this morning (sorry I'm in Taiwan)
bors r+

**Comment by davll** [comment]
Oops, I forgot to fix the PR description before invoking bors...

**Comment by davll** [comment]
@kvark 
It seems that `cargo build` is not able to ignore some crates correctly :(
Thus I've made conditional compilation to explicitly disable incompability compilation

**Comment by kvark** [comment]
@davll I'm working on a build system refactor in #1534

**Pull request merged**